src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java

```diff
@@ -191,6 +197,48 @@ void test_health_reports_success_if_messages_check_disabled_in_current_environme
         verifyNoInteractions(caseEventMessageRepository);
     }
 
+    @ParameterizedTest
+    @MethodSource(value = "enabledEnvProvider")
+    void test_enabled_for_environment_return_false_prod_aat(String env) {
+
+        setField(receivedMessagesHealthController, "receivedMessageCheckEnvEnabled", "prod,aat");
+
+        MockHttpServletRequest mockHttpServletRequest = new MockHttpServletRequest();
+        mockHttpServletRequest.setServerName("case-event-handler.any.environment");
+        mockHttpServletRequest.setRequestURI("/health");
+
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(mockHttpServletRequest));
+
+        assertFalse(receivedMessagesHealthController.isNotEnabledForEnvironment(env));
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "disabledEnvProvider")
+    void test_enabled_for_environment_return_true_demo_ithc(String env) {
+
+        setField(receivedMessagesHealthController, "receivedMessageCheckEnvEnabled", "prod,aat");
+
+        MockHttpServletRequest mockHttpServletRequest = new MockHttpServletRequest();
+        mockHttpServletRequest.setServerName("case-event-handler.any.environment");
```

Comment:
Could the server name every be `null` or empty string?  If it is always guaranteed to be present then that's ok as is but if not then you could add a test where it is empty too.

Reply:
This method is called only in HTTP context. I added an unit test to expect exception its called outside the context .